functions/index.js

```diff
@@ -247,6 +247,18 @@ function createRoomAndJoin(league, roomsList, username, id) {
   return roomID.toString();
 }
 
+function updateUserStats(starIncrease,trophieIncrease,uid) {
+
+  admin.database().ref('users').child(uid).transaction(function(user) {
+  if (user) {
+      user.stars += starIncrease;
+      user.trophies += trophieIncrease;
```

Comment:
I think it's better you update directly the stars and trophies in firebase directly from here. Otherwise the the android client would receive the user object in java and this latter would have to update himself firebase which is not very efficient since you could do it here directly.

Reply:
Sorry, I do not understand what you mean. This is done on the server side

Reply:
Ye sure the function is executed on firebase servers but you do not update firebase database here. You are just doing an addition on server side and return the object. I can try to give you an example. Let's say you implement a function `add(x,y) {return x + y}` on server side. Then client (android device) will call this method and the server will execute the addition but it is not very useful to do this since the addition could already be done in client (android-java code). And this is what you are currently doing here. It would make more sense to do operations like setting or removing values in firebase.

Reply:
Nevermind I did not see you were using a transaction.